core/isolate.rs

```diff
@@ -204,7 +193,7 @@ impl Isolate {
 
     let shared = SharedQueue::new(RECOMMENDED_SIZE);
 
-    let needs_init = true;
+    let needs_init = AtomicBool::new(true);
```

Comment:
Why does this need to be atomic now?

Reply:
So I can run init without a mutable reference, but I just realized this getting checked every time we run `poll` is going to slow things way down.

Reply:
It should be possible to revert this entirely, but I'm going to wait for the time being just in case I need it somewhere else.